src/schema/mbs_types.ts

```diff
@@ -0,0 +1,267 @@
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import _ from 'lodash';
+import debug from 'debug';
+import * as graphql from 'graphql';
+import {
+  GraphQLBoolean,
+  GraphQLInt,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+} from 'graphql';
+import { delegateToSchema } from '@graphql-tools/delegate';
+
+import schema from './schema';
+import { CommitObject, DistGitInstanceInputType } from './distgit_types';
+import {
+  KojiBuildTagsType,
+  KojiHistoryType,
+  KojiInstanceInputType,
+} from './koji_types';
+
+const log = debug('osci:mbs_types');
+
+export interface MbsTaskFields {
+  component: string;
+  id?: number;
+  nvr: string;
+  state: number;
+}
+
+export const MbsTaskType = new GraphQLObjectType<MbsTaskFields, {}>({
+  name: 'MbsTaskType',
+  fields: {
+    component: {
+      description: 'Name of the component comprising the task',
+      type: GraphQLNonNull(GraphQLString),
+    },
+    id: { description: 'ID of the Koji task', type: GraphQLInt },
+    nvr: {
+      description: 'Full NVR of the comprising package',
+      type: GraphQLNonNull(GraphQLString),
+    },
+    state: {
+      description: 'Number of the state the task is currently in',
+      type: GraphQLNonNull(GraphQLInt),
+    },
+  },
+});
+
+export interface MbsBuildFields {
+  commit: any;
+  context: string;
+  id: number;
+  koji_tag: string;
+  name: string;
+  owner: string;
+  scmurl: string;
+  scratch: boolean;
+  stream: string;
+  tasks: MbsTaskFields[];
+  time_completed?: string;
+  version: string;
+}
+
+interface MbsBuildCommitArgs {
+  instance: string;
+}
+
+/*
+ * TOOD: The third type argument should really be `MbsBuildCommitArgs`, but using
+ * that causes a type error further down below in when passing the resolver to the
+ * `resolve` field of `commit` within `MbsBuildType`.
+ */
+const commitResolver: graphql.GraphQLFieldResolver<any, {}, any> = async (
+  parentValue,
+  args,
+  context,
+  info,
+) => {
+  const { scmurl } = parentValue;
+  const { instance } = args;
+  const nameWithCommit = _.last(_.split(scmurl, 'modules/'));
+  const [repoName, sha1] = _.split(nameWithCommit, '?#');
+  log('Getting commit object for %s:%s', repoName, sha1);
+  if (!repoName || !sha1) {
+    return {};
```

Comment:
I would throw here an error.
GraphQL will automatically deliver error to backend.I
It is good to have understanding of un-successfull parse.

Reply:
Changed.